src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs

```diff
@@ -70,13 +70,18 @@ public void Configure(ScriptJobHostOptions options)
 
         private void ConfigureFunctionTimeout(ScriptJobHostOptions options)
         {
-            if (options.FunctionTimeout != null)
+            if (options.FunctionTimeout == null)
```

Comment:
Please update the tests to test this new capability. Tests are in ScriptHostOptionsSetupTests.cs (e.g. Configure_AppliesTimeout). E.g. you could add a new test Configure_AppliesInfiniteTimeout